tests/foreman/api/test_subnet.py

```diff
@@ -497,6 +496,34 @@ def test_positive_list_parameters(self):
         :expectedresults: The satellite should display all the subnet
             parameters
         """
+        parameter = [{'name': gen_string('alpha'),
+                      'value': gen_string('alpha')}]
+        org = entities.Organization().create()
+        loc = entities.Location(organization=[org]).create()
+        org_subnet = entities.Subnet(
+            location=[loc],
+            organization=[org],
+            ipam=u'DHCP',
+            vlanid=gen_string('numeric', 3),
+            subnet_parameters_attributes=parameter,
+        ).create()
+        self.assertEqual(org_subnet.subnet_parameters_attributes[0]['name'],
+                         parameter[0]['name'])
+        self.assertEqual(org_subnet.subnet_parameters_attributes[0]['value'],
+                         parameter[0]['value'])
+        sub_param = entities.Parameter(
+            name=gen_string('alpha'),
+            subnet=org_subnet.id,
+            value=gen_string('alpha')
+        ).create()
+        sub = entities.Subnet(id=org_subnet.id).read()
+        params_list = [(param['name'], param['value'])
+                       for param in sub.subnet_parameters_attributes]
+        self.assertEqual(len(params_list), 2)
+        for tuple_ in params_list:
+            if sub_param.name in tuple_:
```

Comment:
The point here is to get all parameters from subnet and verify that this new one is among them, right?

Perhaps drop this if and filter it out when constructing `params_list`, something like that:
```python
        params_list = [(param['name'], param['value'])
                       for param in sub.subnet_parameters_attributes
                       if param['name'] == sub_param.name]
````

And if `params_list` is a dict, as suggested earlier, you could just do:
```
self.assertEqual(params_list[sub_param.name], sub_param.value)
```
(you could also explicitly assert that `sub_param.name` is among dict keys, but line above will throw exception if it is not, so there is limited value in that)

Reply:
Addressed this suggestion.